src/styles/text/text.js

```diff
@@ -416,6 +414,26 @@ Object.assign(TextStyle, {
         }
 
         return labels;
+    },
+
+    // Build one or more labels for a line geometry
+    buildLineLabels (size, line, options, labels) {
+        let subdiv = Math.min(options.subdiv, line.length);
```

Comment:
Yep that fixes the issue. I haven't run it but the code path seems ok for that case now.

>I think it's more straightforward to keep the different label placements to their own separate segment ranges

That's true it makes sense, and I feel like this would also help to keep the segment spacing more homogeneous.